### PR TITLE
Node Operator guide up-to-date information

### PR DESCRIPTION
**Pull request by kolyasapphire**
Keeping Node Operator tools information and other things updated.

### CODE REVIEW

File: docs/guides/node-operator-manual.md [hunk 1]
```diff
@@ -105,23 +105,43 @@ a Node Operator submits a set of the corresponding signatures [as defined in the
...
+[etherscan page for the mainnet-deployed lido]: https://etherscan.io/address/0xae7ab96520de3a18e5e111b5eaab095312d7fe84#readProxyContract
 [etherscan page for the prater-deployed lido]: https://goerli.etherscan.io/address/0x1643E812aE58766192Cf7D2Cf9567dF2C37e9B7F#readProxyContract
 [`lib/abi/lido.json`]: https://github.com/lidofinance/lido-dao/blob/971ac8f/lib/abi/Lido.json
 
-#### Using the forked eth2.0-deposit-cli
+#### Using eth2.0-deposit-cli
+
+##### Mainnet
+
+For a mainnet deployment, use the latest release of [`eth2.0-deposit-cli`].
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:132]: Is there a way to used pre-defined withdrawal pubkey with the non-forked cli app? The difference between the mainnet and the testnet flow seems a bit off

File: docs/guides/node-operator-manual.md [hunk 2]
```diff
@@ -178,30 +214,44 @@ Etherscan pages for the Görli/Prater contracts:
...
 [`lib/abi/nodeoperatorsregistry.json`]: https://github.com/lidofinance/lido-dao/blob/971ac8f/lib/abi/NodeOperatorsRegistry.json
 [`nodeoperatorsregistry.getnodeoperator`]: https://github.com/lidofinance/lido-dao/blob/971ac8f/contracts/0.4.24/nos/NodeOperatorsRegistry.sol#L335
 
-#### Using the key submitter UI
+#### Using the batch key submitter UI
+
+Lido has specialised key submitters: [Mainnet web interface for submitting the keys] and a [Testnet web interface for submitting the keys].
 
-Lido has the specialized [web interface for submitting the keys].
+It will require that all keys are validated first - click the check button. Please be patient as validation can take a long time. Also, please don't update the page - UI has a built-in request failure retries integrated and will display an error if all request fail.
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:231]: 1. What is the ballpark of a long time here? It's about 10 minutes for 100 keys, right?
2. `built-in request failure retries integrated` — that's a bit hard to parse. Seems like something like "the page repeats the request if the current check has failed, and will show an error after couple of retries" would be a bit easier to read.

File: docs/guides/node-operator-manual.md [hunk 3]
```diff
@@ -276,3 +326,35 @@ docker run --rm -it \
   --datadir /root/.lighthouse/data \
   --directory /root/validator_keys
 ```
+
+### Checking the keys of all Lido Node Operators
+
+After keys are submitted, it's encouraged to wait for the transactions to be mined and to check how they appear on the chain.
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:332]: "Please, check the submitted keys before starting the vote on increasing the key limit."

File: docs/guides/node-operator-manual.md [hunk 4]
```diff
@@ -276,3 +326,35 @@ docker run --rm -it \
...
+#### Lido CLI
+
+Make sure Python with pip is installed and then run:
+
+```sh
+pip install lido-cli
+lido-cli --rpc http://1.2.3.4:8545 validate_network_keys
+```
+
+This operation checks every Lido key for validity and thus is very CPU-intensive and takes a long time to process the keys.
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:343]: "Note that the check is a CPU-heavy operation and could take some time to process the keys."
If we may give the ballpark for the time of check (minutes, hours?) — would be great

File: docs/guides/node-operator-manual.md [hunk 5]
```diff
@@ -276,3 +326,35 @@ docker run --rm -it \
...
+
+You would need an RPC endpoint - a local node / RPC provider (eg Alchemy/Infura).
+
+If the new keys are present and valid, Node Operators can vote for increasing the key limit for the Node Operator.
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:347]: Need a note on what to do if there's something wrong and keys aren't valid

File: docs/guides/node-operator-manual.md (hunk 1 above)
Comment (kolyasapphire) [reply docs/guides/node-operator-manual.md:132]: I don't think there is - we actually forked it for this exact reason :)

Non-forked eth2-deposit-cli can take only `eth1_withdrawal_address` as an option, otherwise it will use a mnemonic-derived withdrawal pubkey.

Comment (kolyasapphire) [reply docs/guides/node-operator-manual.md:132]: And we don't really have a clean way to get the withdrawal address for node ops. Our contract has a view function to get the whole withdrawal_credentials with withdrawal address embedded, thus need easy but manual extraction. This is far from ideal, I completely agree, but can't think of any way to improve this.

File: docs/guides/node-operator-manual.md (hunk 3 above)
Comment (kolyasapphire) [reply docs/guides/node-operator-manual.md:332]: I don't think this is explicit enough. I suggest this:
```
### Checking the keys of all Lido Node Operators

After keys are submitted, it's critical to first wait for the key submission transactions to be mined and to verify them with all Lido keys after they appear on the blockchain.

Never vote for increasing the key limits of Node Operators before verifying new keys are present and valid.
```

File: docs/guides/node-operator-manual.md (hunk 5 above)
Comment (kolyasapphire) [reply docs/guides/node-operator-manual.md:347]: Agreed, adding this

File: docs/guides/node-operator-manual.md (hunk 2 above)
Comment (kolyasapphire) [reply docs/guides/node-operator-manual.md:231]: 1. No, much faster, but honestly we currently don't have even an idea of the processing time after our checkers will be scaled - better wait and see. We can add something vague though - eg 1k keys under a minute, but it doesn't at all account for simultaneous checks - we don't have a queue implemented
2. Agree, will change the wording

File: docs/guides/node-operator-manual.md [hunk 6]
```diff
@@ -276,3 +326,43 @@ docker run --rm -it \
...
+
+#### Lido Node Operator Dashboard
+
+You can also check the uploaded keys on [Mainnet Lido Node Operator Dashboard] or [Testnet Lido Node Operator Dashboard].
+
+It will display key summary for each node operator, including checks for number of keys over staking limit and invalid keys if found.
+
+It is updated every 30 minutes via cron, but update period may change in the future.
+
+[mainnet lido node operator dashboard]: https://stake.lido.fi/key-checker/existing
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:357]: Need to update the links here

File: docs/guides/node-operator-manual.md [hunk 7]
```diff
@@ -276,3 +326,43 @@ docker run --rm -it \
...
+
+This operation checks all Lido keys for validity. This is a CPU-intensive process, for example, a modern desktop with 6 cores, 12 threads and great cooling processes 1k keys in ~1 minute.
+
+You would need an RPC endpoint - a local node / RPC provider (eg Alchemy/Infura).
+
+#### Lido Node Operator Dashboard
+
+You can also check the uploaded keys on [Mainnet Lido Node Operator Dashboard] or [Testnet Lido Node Operator Dashboard].
+
+It will display key summary for each node operator, including checks for number of keys over staking limit and invalid keys if found.
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:353]: "The UI show number of submitted, approved and valid keys for each Node Operator, along with all invalid keys in case there are any."

File: docs/guides/node-operator-manual.md [hunk 8]
```diff
@@ -276,3 +326,43 @@ docker run --rm -it \
   --datadir /root/.lighthouse/data \
   --directory /root/validator_keys
 ```
+
+### Checking the keys of all Lido Node Operators
+
+After keys are submitted, it's critical to first wait for the key submission transactions to be mined and after that check all Lido keys for validity.
+
+Never vote for increasing the key limits of Node Operators before verifying new keys are present and valid.
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:334]: Please, check there is enough valid keys submitted before voting for key limit increase.

File: docs/guides/node-operator-manual.md [hunk 9]
```diff
@@ -276,3 +326,43 @@ docker run --rm -it \
   --datadir /root/.lighthouse/data \
   --directory /root/validator_keys
 ```
+
+### Checking the keys of all Lido Node Operators
+
+After keys are submitted, it's critical to first wait for the key submission transactions to be mined and after that check all Lido keys for validity.
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:332]: The tooling for the key checking works only with the on-chain data. Make sure to get the key submission transactions confirmed before checking the keys.

File: docs/guides/node-operator-manual.md [hunk 10]
```diff
@@ -210,17 +260,17 @@ If you’ve used the forked `eth2.0-deposit-cli`, you can paste the content of t
 
 Click `Check` button, and then the interface would run required checks connect the MetaMask and click `Submit` button.
 
-Once the keys are submitted, Node Operators can check whether the supplied keys are valid with the [web interface for checking the submitted keys]. If the keys are valid, they can vote for increasing the key limit for the Node Operator.
-
-- [web interface for submitting the keys](https://stake.testnet.lido.fi/key-checker/submit)
-- [web interface for checking the submitted keys](https://stake.testnet.lido.fi/key-checker/existing)
+[mainnet web interface for submitting the keys]: https://stake.lido.fi/key-checker/submit
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:263]: Need the correct link here

File: docs/guides/node-operator-manual.md [hunk 11]
```diff
@@ -149,13 +169,29 @@ fields except `pubkey` and `signature` from the array items.
 Never share the generated mnemonic and your private keys with anyone, including the protocol members
 and DAO holders.
 
+[`eth2.0-deposit-cli`]: https://github.com/ethereum/eth2.0-deposit-cli/releases
 [a fork of `eth2.0-deposit-cli`]: https://github.com/lidofinance/eth2.0-deposit-cli
 [`lidofinance/deposit-cli`]: https://hub.docker.com/repository/docker/lidofinance/deposit-cli
 
+### Validating the keys
+
+It's vital to check the correctness of the keys before submitting them.
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:178]: Please, make sure to check the keys validity before submitting them on-chain.

File: docs/guides/node-operator-manual.md [hunk 12]
```diff
@@ -149,13 +169,29 @@ fields except `pubkey` and `signature` from the array items.
...
+
+If you will be submitting keys using Lido's submitter, they will be checked before submitting - the "submit" button won't be active unless keys are validated and they are valid.
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:180]: Lido submitter tooling checks the keys automatically.

File: docs/guides/node-operator-manual.md [hunk 13]
```diff
@@ -149,13 +169,29 @@ fields except `pubkey` and `signature` from the array items.
...
+
+If you will be submitting keys manually via Lido contract, you can use Lido CLI, it's as simple as having Python with pip installed and running:
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:182]: If you will be submitting keys manually via Lido contract, you can use Lido CLI. 
It's distributed as a pip package:

File: docs/guides/node-operator-manual.md [hunk 14]
```diff
@@ -178,30 +214,44 @@ Etherscan pages for the Görli/Prater contracts:
...
+
 [`addsigningkeysoperatorbh` function]: https://github.com/lidofinance/lido-dao/blob/971ac8f/contracts/0.4.24/nos/NodeOperatorsRegistry.sol#L250
 [`getoperators()` function]: https://github.com/lidofinance/lido-dao/blob/971ac8f/contracts/0.4.24/Lido.sol#L361
 [`lib/abi/nodeoperatorsregistry.json`]: https://github.com/lidofinance/lido-dao/blob/971ac8f/lib/abi/NodeOperatorsRegistry.json
 [`nodeoperatorsregistry.getnodeoperator`]: https://github.com/lidofinance/lido-dao/blob/971ac8f/contracts/0.4.24/nos/NodeOperatorsRegistry.sol#L335
 
-#### Using the key submitter UI
+#### Using the batch key submitter UI
+
+Lido has specialised key submitters: [Mainnet web interface for submitting the keys] and a [Testnet web interface for submitting the keys].
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:229]: Lido provides UIs for key submission: ...

File: docs/guides/node-operator-manual.md [hunk 15]
```diff
@@ -178,30 +214,44 @@ Etherscan pages for the Görli/Prater contracts:
...
+
+It will require that all keys are validated first - click the check button. Please be patient - we aim to process 1k keys under a minute, but several users can be checking keys simultaneously, so the process can take longer. Also, please don't update the page manually - UI will automatically repeat failing requests and will show an error if retries fail.
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:231]: Please, validate the keys before submission by clicking the `check` button. We're aiming to check 1k keys / minute, so this could take some time. Please, don't refresh the page during the checks — this would just start the process over from the very beginning. UI will retry failing requests if there are any errors, and will show an error directly if retries fail.

File: docs/guides/node-operator-manual.md (hunk 6 above)
Comment (kolyasapphire) [reply docs/guides/node-operator-manual.md:357]: Yep, good catch, new deployments will be live soon, updating

File: docs/guides/node-operator-manual.md [hunk 16]
```diff
@@ -105,23 +105,43 @@ a Node Operator submits a set of the corresponding signatures [as defined in the
 The fork version used for generating the signature must correspond to the fork version of the Beacon
 chain the instance of Lido protocol is targeted to.
 
+#### Mainnet
+
+Make sure to obtain a correct withdrawal address by finding it inside active withdrawal credentials either on Aragon UI or calling the contract via [`Lido.getWithdrawalCredentials()`]. You can quickly find on the [Etherscan page for the Mainnet-deployed Lido].
```
Comment (kadmil) [inline docs/guides/node-operator-manual.md:110]: inside _the_ active ...
You can find the method on the ... (no `quickly`, add `method`)

**Review by kadmil — APPROVED** [review]
LGTM; outstanding comments are minor cosmetics
